Modules/Registration/Common/include/itkImageToImageMetric.hxx

```diff
@@ -457,15 +456,13 @@ ImageToImageMetric<TFixedImage, TMovingImage>::MultiThreadingInitialize()
     }
     else
     {
-      this->m_BSplineTransformWeights.SetSize(this->m_NumBSplineWeights);
       this->m_BSplineTransformIndices.SetSize(this->m_NumBSplineWeights);
```

Comment:
Could this variable receive the same constant treatment?

Reply:
> Could this variable receive the same constant treatment?

@dzenanz Indeed! I think that `m_BSplineTransformIndices` could also be declared as `FixedArray` (rather than `Array`), which would probably yield even more performance gain  😃  Would you want to have it included with the very same pull request, or would it be OK to do that in another PR?

Reply:
I prefer it done as part of this PR.

Reply:
@dzenanz Added another commit to this PR, to declare the indices as FixedArray: "PERF: Use FixedArray for BSplineBaseTransform ParameterIndexArrayType". Indeed, even more PERF gain! 😃